src/controller/sms/signup/signup_steps.js

```diff
@@ -0,0 +1,221 @@
+/* eslint-disable no-param-reassign */
+const message = require('../message')
+const saveUser = require('../../../user/save_user')
+const issues = require('../../../public/issues').issues
+const events = require('../schedule')
+const debug = require('debug')('sms')
+
+function notStarted(req, step) {
+  return !req.session.steps[step]
+}
+function markStarted(req, step) {
+  req.session.steps[step] = 'started'
+}
+function markNotStarted(req, step) {
+  delete req.session.steps[step]
+}
+function markComplete(req, step) {
+  req.session.steps[step] = 'complete'
+}
+
+const validZip = /^\d{5}$/
+const startsWithY = /^y/i
+const containsYes = /(yes|yeah|ya)/i
+
+function isAffirmative(response) {
+  if (startsWithY.test(response)) return true
+  if (containsYes.test(response)) return true
+  return false
+}
+
+function validateZip(zip) {
+  return validZip.test(zip) && zip
```

Comment:
this function returns the zip if it's valid, or false if it's not. so the `&& zip` just changes a `true` return value to the zip itself.